Content.Server/GameObjects/Components/Explosion/ClusterFlashComponent.cs

```diff
@@ -0,0 +1,180 @@
+#nullable enable
+using Content.Shared.Interfaces.GameObjects.Components;
+using Content.Server.GameObjects.Components.Explosion;
+using Robust.Shared.GameObjects;
+using System.Threading.Tasks;
+using Robust.Server.GameObjects.Components.Container;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Serialization;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Content.Server.GameObjects.Components.Trigger.TimerTrigger;
+using Content.Server.Throw;
+using Robust.Server.GameObjects;
+using Content.Shared.GameObjects.Components.Explosion;
+using Content.Shared.Physics;
+using Robust.Shared.GameObjects.Components;
+using Robust.Shared.GameObjects.Components.Timers;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Random;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Explosives
+{
+    [RegisterComponent]
+    public sealed class ClusterFlashComponent : Component, IInteractUsing, IUse
+    {
+        public override string Name => "ClusterFlash";
+
+        private Container _grenadesContainer = default!;
+
+        /// <summary>
+        ///     What we fill our prototype with if we want to pre-spawn with grenades.
+        /// </summary>
+        [ViewVariables]
+        private string? _fillPrototype;
+
+        /// <summary>
+        ///     Maximum grenades in the container.
+        /// </summary>
+        [ViewVariables]
+        private int _maxGrenades;
+
+        [ViewVariables]
+        private int _maxViewableGrenades;
+
+        /// <summary>
+        ///     How long until our grenades are shot out and armed.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        private float _delay;
+
+        /// <summary>
+        ///     Max distance grenades can be thrown.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        private float _throwDistance;
+
+        /// <summary>
+        ///     This is the end.
+        /// </summary>
+        private bool _countDown;
+
+        // I'm suss on this bit
+        async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs args)
+        {
+            if (_grenadesContainer.ContainedEntities.Count >= _maxGrenades || !args.Using.HasComponent<FlashExplosiveComponent>())
+                return false;
+
+            _grenadesContainer.Insert(args.Using);
+            UpdateAppearance();
+            return true;
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _fillPrototype, "fillPrototype", null);
+            serializer.DataField(ref _maxGrenades, "maxGrenadesCount", 4);
```

Comment:
3 != 4

Reply:
I don't know why you don't like it. I really should to "link" sprite to grenades count? I was thinking like: "Hey, there will be is fourth grenade at the back!"

Reply:
The visual for the cluster only shows 3. It'll be confusing to people if they don't realize there's another slot they could have been using but there was no visual queue. You should make it 3.

Reply:
Now i use RoundToLevels. Not sure if it is confusing when you put third grenade and nothing changes but fourth adds last. What do you think, @metalgearsloth, @SweptWasTaken?